src/hooks/usePlayground.js

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from 'react';
+import state from '../lib/state';
+import { useAppContext } from '../components/Context';
+import parser from '../parser';
+
+function usePlayground({ initialMarkup, initialQuery }) {
+  const [markup, setMarkup] = useState(initialMarkup);
+  const [query, setQuery] = useState(initialQuery);
+  const { setParsed, htmlRoot } = useAppContext();
+
+  useEffect(() => {
+    if (!htmlRoot) {
+      return;
+    }
+
+    const parsed = parser.parse({ htmlRoot, query });
+    setParsed(parsed);
+
+    state.updateTitle(parsed.expression?.expression);
```

Comment:
This shouldn't be here, it should be next to the `save state` call. Multiple playgrounds will now all modify the window title.